EventFilter/Phase2PixelRawToDigi/test/plugins/Phase2PixelQCoreNtuple.cc

```diff
@@ -0,0 +1,971 @@
+
+  pixeltreeOnTrack_->Branch("trkPt", &trkPt_);
+  pixeltreeOnTrack_->Branch("trkEta", &trkEta_);
+  pixeltreeOnTrack_->Branch("trkTheta", &trkTheta_);
+  pixeltreeOnTrack_->Branch("trkPhi", &trkPhi_);
+  pixeltreeOnTrack_->Branch("trkIsHighPurity", &trkIsHighPurity_);
+  pixeltreeOnTrack_->Branch("cotAlphaFromTrack", &recHit_.cotAlphaFromTrack, "cotAlphaFromTrack/F");
+  pixeltreeOnTrack_->Branch("cotBetaFromTrack", &recHit_.cotBetaFromTrack, "cotBetaFromTrack/F");
+
+  pixeltreeOnTrack_->Branch("DgN", &recHit_.fDgN, "DgN/I");
+  pixeltreeOnTrack_->Branch("DgRow", recHit_.fDgRow, "DgRow[DgN]/I");
+  pixeltreeOnTrack_->Branch("DgCol", recHit_.fDgCol, "DgCol[DgN]/I");
+  pixeltreeOnTrack_->Branch("DgDetId", recHit_.fDgDetId, "DgDetId[DgN]/I");
+  pixeltreeOnTrack_->Branch("DgAdc", recHit_.fDgAdc, "DgAdc[DgN]/F");
+  pixeltreeOnTrack_->Branch("DgCharge", recHit_.fDgCharge, "DgCharge[DgN]/F");
+}
+
+// Functions that gets called by framework every event
+void Phase2PixelQCoreNtuple::analyze(const edm::Event& e, const edm::EventSetup& es) {
+  //Retrieve tracker topology from geometry
+  const TrackerTopology* const tTopo = &es.getData(tTopoToken_);
+
+  // geometry setup
+  const TrackerGeometry* const theGeometry = &es.getData(geomToken_);
+
+  std::vector<PSimHit> matched;
+  const PSimHit* closest_simhit = nullptr;
+
+  edm::Handle<SiPixelRecHitCollection> recHitColl;
+  e.getByToken(pixelRecHits_token_, recHitColl);
+
+  edm::Handle<edm::DetSetVector<Phase2ITQCore>> aQCoreVector;
+  e.getByToken(qcore_token_, aQCoreVector);
+
+  edm::Handle<edm::DetSetVector<Phase2ITChipBitStream>> aBitStreamVector;
+  e.getByToken(bitstream_token_, aBitStreamVector);
+
+  edm::DetSetVector<Phase2ITQCore>::const_iterator iterDet;
+  for (iterDet = aQCoreVector->begin(); iterDet != aQCoreVector->end(); iterDet++) {
+    DetId tkId = iterDet->id;
+
+    edm::DetSet<Phase2ITQCore> theQCores = (*aQCoreVector)[tkId];
+
+    //std::cout << "QCORE DETID NEW : " << tkId.rawId() << std::endl;
+
+    for (auto iterQCore = theQCores.begin(); iterQCore != theQCores.end(); ++iterQCore) {
+    }
+  }
+
+  edm::DetSetVector<Phase2ITChipBitStream>::const_iterator iterDetBitStream;
+  for (iterDetBitStream = aBitStreamVector->begin(); iterDetBitStream != aBitStreamVector->end(); iterDetBitStream++) {
+    DetId tkId = iterDetBitStream->id;
+
+    edm::DetSet<Phase2ITChipBitStream> theBitStreams = (*aBitStreamVector)[tkId];
+
+    //std::cout << "BITSTREAM DETID : " << tkId.rawId() << std::endl;
+
+    for (auto iterBitStream = theBitStreams.begin(); iterBitStream != theBitStreams.end(); ++iterBitStream) {
+      //std::cout << "BITSTREAM : " << iterBitStream->get_rocid() << " size = " << iterBitStream->get_bitstream().size() << std::endl;
+    }
+  }
+
+  // for finding matched simhit
+  TrackerHitAssociator associate(e, trackerHitAssociatorConfig_);
+
+  edm::Handle<edm::DetSetVector<PixelDigi>> pixelDigiHandle;
+  e.getByToken(pixelDigi_token_, pixelDigiHandle);
+
+  /*
```

Comment:
Is there a good reason to add this commented out code block to CMSSW?

Reply:
Hi I cleaned up the unnecessary bits